File: backend/src/main/kotlin/com/mongle/controller/UserController.kt
package com.mongle.controller

import com.mongle.common.context.AuthUser
import com.mongle.common.context.UserPrincipal
import com.mongle.controller.dto.UserProfileRequest
import com.mongle.controller.dto.UserProfileResponse
import com.mongle.service.UserService
import io.swagger.v3.oas.annotations.Operation
import io.swagger.v3.oas.annotations.tags.Tag
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.DeleteMapping
import org.springframework.web.bind.annotation.PatchMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.ResponseStatus
import org.springframework.web.bind.annotation.RestController

@Tag(name = "사용자", description = "인증된 사용자를 관리한다.")
@RestController
@RequestMapping("/api/v1/users/me")
class UserController(
    private val userService: UserService,
) {
    @Operation(
        summary = "최초 프로필 설정 완료",
        description = "기본 아바타 또는 업로드 이미지를 저장한다. 값을 보내지 않으면 프로필 설정을 건너뛴다.",
    )
    @PatchMapping("/profile")
    fun completeProfileSetup(
        @AuthUser user: UserPrincipal,
        @RequestBody request: UserProfileRequest,
    ): UserProfileResponse = userService.completeProfileSetup(user.id, request)

    @Operation(
        summary = "현재 사용자 삭제",
        description = "테스트를 위해 현재 사용자를 삭제한다.",
    )
    @DeleteMapping
    @ResponseStatus(HttpStatus.NO_CONTENT)
    fun deleteCurrentUser(
        @AuthUser user: UserPrincipal,
    ) = userService.deleteCurrentUser(user.id)
}

File: backend/src/users/user.controller.ts
import { Body, Controller, Delete, HttpCode, HttpStatus, Patch, UseGuards } from '@nestjs/common';
import { ApiBearerAuth, ApiNoContentResponse, ApiOkResponse, ApiOperation, ApiTags } from '@nestjs/swagger';
import { CurrentUser } from '../common/auth/current-user.decorator';
import { JwtAuthGuard } from '../common/auth/jwt-auth.guard';
import { UserPrincipal } from '../common/auth/user-principal.interface';
import { UserProfileRequest, UserProfileResponse } from './user.dto';
import { UserService } from './user.service';

@ApiTags('사용자')
@ApiBearerAuth('bearerAuth')
@UseGuards(JwtAuthGuard)
@Controller('api/v1/users/me')
export class UserController {
  constructor(private readonly userService: UserService) {}

  @Patch('profile')
  @HttpCode(HttpStatus.OK)
  @ApiOperation({
    operationId: 'completeProfileSetup',
    summary: '최초 프로필 설정 완료',
    description: '기본 아바타 또는 업로드 이미지를 저장한다. 값을 보내지 않으면 프로필 설정을 건너뛴다.',
  })
  @ApiOkResponse({ type: UserProfileResponse })
  completeProfileSetup(
    @CurrentUser() user: UserPrincipal,
    @Body() request: UserProfileRequest,
  ): Promise<UserProfileResponse> {
    return this.userService.completeProfileSetup(user.id, request);
  }

  @Delete()
  @HttpCode(HttpStatus.NO_CONTENT)
  @ApiOperation({
    operationId: 'deleteCurrentUser',
    summary: '현재 사용자 삭제',
    description: '테스트를 위해 현재 사용자를 삭제한다.',
  })
  @ApiNoContentResponse()
  async deleteCurrentUser(@CurrentUser() user: UserPrincipal): Promise<void> {
    await this.userService.deleteCurrentUser(user.id);
  }
}
